src/util/key.rs

```diff
@@ -197,6 +198,118 @@ impl ops::Index<ops::RangeFull> for PrivateKey {
     }
 }
 
+#[cfg(feature = "serde")]
+impl ::serde::Serialize for PrivateKey {
+    fn serialize<S: ::serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        s.collect_str(self)
```

Comment:
Hmm. I didn't know about this method. Would be nice if there was an equivalent for just using `FromStr` in the Deserialize.

Reply:
Well, unfortunately `FromStr` allocates..